File: scripts/svg-to-png.js
#!/usr/bin/env node
/**
 * Convert SVG logo to PNG for Stripe branding.
 * Stripe requires: square, at least 128px, max 512KB, PNG or JPG format.
 */

const fs = require('fs');
const path = require('path');

function convertWithResvg(svgPath, pngPath, size) {
    try {
        const { Resvg } = require('@resvg/resvg-js');
        const svgContent = fs.readFileSync(svgPath, 'utf-8');
        
        const resvg = new Resvg(svgContent, {
            fitTo: {
                mode: 'width',
                value: size,
            },
        });
        
        const pngData = resvg.render();
        const pngBuffer = pngData.asPng();
        
        fs.writeFileSync(pngPath, pngBuffer);
        return Promise.resolve();
    } catch (e) {
        return null;
    }
}

async function convertSvgToPng(svgPath, pngPath, size) {
    const result = convertWithResvg(svgPath, pngPath, size);
    if (result) {
        return result;
    }
    
    throw new Error('No SVG conversion library found. Install one: npm install --save-dev @resvg/resvg-js');
}

async function main() {
    const scriptDir = __dirname;
    const projectRoot = path.dirname(scriptDir);
    const assetsDir = path.join(projectRoot, 'assets');
    
    const svgFiles = [
        { svg: 'logo-alttext-ai.svg', sizes: [128, 512] },
        { svg: 'logo-alttext-ai-white-bg.svg', sizes: [128, 512] }
    ];
    
    console.log('🔄 Converting SVG logos to PNG...\n');
    
    let successCount = 0;
    
    for (const { svg, sizes } of svgFiles) {
        const svgPath = path.join(assetsDir, svg);
        
        if (!fs.existsSync(svgPath)) {
            console.log(`⚠️  Skipping ${svg} (file not found)`);
            continue;
        }
        
        for (const size of sizes) {
            const baseName = path.basename(svg, '.svg');
            const pngName = `${baseName}-${size}x${size}.png`;
            const pngPath = path.join(assetsDir, pngName);
            
            try {
                process.stdout.write(`   Converting ${svg} → ${pngName} (${size}x${size})... `);
                
                await convertSvgToPng(svgPath, pngPath, size);
                
                const stats = fs.statSync(pngPath);
                const fileSizeKB = (stats.size / 1024).toFixed(1);
                
                console.log(`✅ (${fileSizeKB} KB)`);
                successCount++;
            } catch (error) {
                console.log(`❌ Failed: ${error.message}`);
            }
        }
    }
    
    console.log('');
    if (successCount > 0) {
        console.log(`✅ Successfully created ${successCount} PNG file(s)`);
        console.log('');
        console.log('📋 Stripe-ready logo files:');
        console.log('   Recommended: assets/logo-alttext-ai-512x512.png');
        console.log('   Minimum size: assets/logo-alttext-ai-128x128.png');
    } else {
        console.log('❌ No PNG files were created');
        console.log('');
        console.log('💡 To install conversion library, run:');
        console.log('   npm install --save-dev sharp');
        process.exit(1);
    }
}

main().catch(error => {
    console.error('Error:', error.message);
    process.exit(1);
});

File: scripts/svg-to-png.py
#!/usr/bin/env python3
"""
Convert SVG logo to PNG for Stripe branding.
Stripe requires: square, at least 128px, max 512KB, PNG or JPG format.
"""

import sys
import os
import subprocess

def install_cairosvg():
    """Install cairosvg if not available."""
    try:
        import cairosvg
        return True
    except ImportError:
        print("Installing cairosvg...")
        try:
            subprocess.check_call([sys.executable, "-m", "pip", "install", "cairosvg", "--quiet"])
            import cairosvg
            return True
        except Exception as e:
            print(f"Failed to install cairosvg: {e}")
            return False

def convert_svg_to_png(svg_path, png_path, size=512):
    """Convert SVG to PNG at specified size."""
    try:
        import cairosvg
        cairosvg.svg2png(url=svg_path, write_to=png_path, output_width=size, output_height=size)
        return True
    except Exception as e:
        print(f"Error converting {svg_path}: {e}")
        return False

def main():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    project_root = os.path.dirname(script_dir)
    assets_dir = os.path.join(project_root, "assets")
    
    if not install_cairosvg():
        print("\n❌ Could not install cairosvg. Please install manually:")
        print("   pip install cairosvg")
        sys.exit(1)
    
    # Convert both SVG files to PNG
    svg_files = [
        ("logo-alttext-ai.svg", "logo-alttext-ai-512x512.png"),
        ("logo-alttext-ai-white-bg.svg", "logo-alttext-ai-white-bg-512x512.png"),
    ]
    
    # Also create 128px versions (minimum required)
    svg_files.extend([
        ("logo-alttext-ai.svg", "logo-alttext-ai-128x128.png"),
        ("logo-alttext-ai-white-bg.svg", "logo-alttext-ai-white-bg-128x128.png"),
    ])
    
    print("🔄 Converting SVG logos to PNG...")
    print("")
    
    success_count = 0
    for svg_name, png_name in svg_files:
        svg_path = os.path.join(assets_dir, svg_name)
        png_path = os.path.join(assets_dir, png_name)
        
        if not os.path.exists(svg_path):
            print(f"⚠️  Skipping {svg_name} (file not found)")
            continue
        
        # Determine size from filename
        if "128x128" in png_name:
            size = 128
        else:
            size = 512
        
        print(f"   Converting {svg_name} → {png_name} ({size}x{size})")
        
        if convert_svg_to_png(svg_path, png_path, size):
            file_size = os.path.getsize(png_path) / 1024  # KB
            print(f"      ✅ Created ({file_size:.1f} KB)")
            success_count += 1
        else:
            print(f"      ❌ Failed")
    
    print("")
    if success_count > 0:
        print(f"✅ Successfully created {success_count} PNG file(s)")
        print("")
        print("📋 Stripe-ready logo files:")
        print("   Recommended: assets/logo-alttext-ai-512x512.png")
        print("   Minimum size: assets/logo-alttext-ai-128x128.png")
    else:
        print("❌ No PNG files were created")
        sys.exit(1)

if __name__ == "__main__":
    main()

